encoding/protobuf/error.go

```diff
@@ -170,25 +170,29 @@ func createStatusWithDetail(pberr *pberror, encoding transport.Encoding, codec *
 }
 
 func setApplicationErrorMeta(pberr *pberror, resw transport.ResponseWriter) {
-	applicationErroMetaSetter, ok := resw.(transport.ApplicationErrorMetaSetter)
+	applicationErrorMetaSetter, ok := resw.(transport.ApplicationErrorMetaSetter)
 	if !ok {
 		return
 	}
 
-	decodedDetails := GetErrorDetails(pberr)
-	var appErrName string
-	if len(decodedDetails) > 0 { // only grab the first name since this will be emitted with metrics
-		appErrName = messageNameWithoutPackage(proto.MessageName(
-			decodedDetails[0].(proto.Message)),
-		)
-	}
+	var (
+		decodedDetails = GetErrorDetails(pberr)
+
+		appErrName = ""
+		details    = make([]string, 0, len(decodedDetails))
+	)
 
-	details := make([]string, 0, len(decodedDetails))
 	for _, detail := range decodedDetails {
-		details = append(details, protobufMessageToString(detail.(proto.Message)))
+		if m, ok := detail.(proto.Message); ok {
```

Comment:
Previously it was assumed that decodedDetails contains only proto.Message, which is may be false sometimes (see unit-test), so here I added check to be sure for 100% that it's proto.Message.